### DISCUSSION TITLE
WASM binaries for meshoptimizer

### DISCUSSION THREAD

**Discussion by sanjeetsuhag** [zeux/meshoptimizer#321] [Q&A] [answered]
I was curious about why the WebAssembly files in the JavaScript transcoders are encoded as base64 in the JS file? Is there any particular benefit or is it just for ease of distribution/integration?

**Comment by zeux** [comment, marked as answer]
It's for ease of distribution as well as faster cold start - it removes a dependent web request. I should mention that based on how caching is implemented today in browsers I wouldn't recommend this for large binaries - it only works well here because the binaries are pretty small (<10KB).

Also, important point - this isn't Base64 :) It's a special encoding optimized for compressibility of the final result. Base64 files are larger *and* compress much worse. See https://github.com/zeux/meshoptimizer/pull/141 and https://github.com/zeux/meshoptimizer/pull/73#issuecomment-550808363
